src/lib/y2storage/initial_guided_proposal.rb

```diff
@@ -255,32 +282,21 @@ def groups_of_candidate_devices
 
     # Sorted list of disks to be tried as root device.
     #
-    # If the current settings already specify a root_device, the list will contain only that one.
+    # In the theoretical case in which the current settings already specify a root_device (in practice,
+    # the initial guided proposal never receives such settings), the list will contain only that device.
     #
-    # Otherwise, it will contain all the candidate devices, sorted from bigger to smaller disk size.
+    # Otherwise, it will contain all the candidate devices, sorted in the order they should be tried.
     #
     # @return [Array<String>]
     def candidate_roots
       return [settings.root_device] if settings.explicit_root_device
 
+      # We use #explicit_candidate_devices because it contains all the devices of the candidate_group.
+      # That's not always true for #candidate_devices. If there are fewer proposed_volumes_sets than
+      # devices in the group, using #candidate_devices could result in giving up before trying all the
+      # possibilities.
```

Comment:
Just for the record, this is the missing explanation in the commit https://github.com/yast/yast-storage-ng/commit/61521afeaf85da19af618855c0c7dd0ceec16e77, introduced as part of #953.

Thanks @ancorgs!